xds/internal/client/client_requests_counter_test.go

```diff
@@ -0,0 +1,161 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package client
+
+import (
+	"sync"
+	"sync/atomic"
+	"testing"
+)
+
+type counterTest struct {
+	name          string
+	maxRequests   uint32
+	numRequests   uint32
+	errorExpected bool
+}
+
+var tests = []counterTest{
+	{
+		name:          "does-not-exceed-max-requests",
+		maxRequests:   1024,
+		numRequests:   1024,
+		errorExpected: false,
+	},
+	{
+		name:          "exceeds-max-requests",
+		maxRequests:   32,
+		numRequests:   64,
+		errorExpected: true,
+	},
+}
+
+func counterTestInit() {
```

Comment:
Why do you need this? Is this to reset the counters? If so, call it `resetServiceRequestsCounter`?

Reply:
Made this change.